src/Akeneo/Pim/Structure/Bundle/Doctrine/ORM/Repository/InternalApi/AttributeSearchableRepository.php

```
@@ -162,6 +168,27 @@ protected function findBySearchQb($search, array $options)
             $qb->setParameter('useable_as_grid_filter', $options['useable_as_grid_filter']);
         }
 
+        if (null !== $options['code']) {
+            switch ($options['code']['type']) {
+                case '1':
+                    $qb->andWhere('a.code LIKE :code');
+                    $qb->setParameter('code', '%'.$options['code']['value'].'%');
+                    break;
+                case '2':
+                    $qb->andWhere('a.code NOT LIKE :code');
+                    $qb->setParameter('code', '%'.$options['code']['value'].'%');
+                    break;
+                case '3':
+                    $qb->andWhere('a.code = :code');
+                    $qb->setParameter('code', $options['code']['value']);
+                    break;
+                case '4':
+                    $qb->andWhere('a.code LIKE :code');
+                    $qb->setParameter('code', $options['code']['value'].'%');
+                    break;
+            }
+        }
```

We should throw an error when the operator is not supported: instead of deleting too much attribute it's better to throw an error in case a filter is added.